source/mir/cv/core/simd.d

```diff
@@ -0,0 +1,77 @@
+/+
+    Utilities to help use SIMD vectors.
++/
+
+module mir.cv.core.simd;
+
+import cpuid.x86_any;
+import std.traits : isNumeric, TemplateOf;
+
+extern(C) @system nothrow @nogc:
+
+private
+{
+    // TODO: Do we need this?
+    struct Init_cpu_id
+    {
+        static void init()
+        {
+            cpuid_x86_any_init();
+        }
+    }
+    static Init_cpu_id __cpuid_init;
+}
+
+
+/++
+    SIMD instructionset traits.
++/
+mixin template Instruction_set_trait(size_t _bitsize, T)
```

Comment:
I guess you're right. `InstructionSet` was actually an idea to cheat, where one function would be used for vectorized and non-vectorized code (by using [Non_SIMD](https://github.com/libmir/mir-cv/blob/imfilter/source/mir/cv/imgproc/imfilter.d#L128)). But I agree it feels cryptic... I'll try redesigning it.